Fiber_Drawing_Tower/Arduino_and_Python_code/GUI.py

```diff
@@ -0,0 +1,299 @@
+
+
+# Creation of green Start button
+button_start = tk.Button(cabestan_frame, text="Start", command=start_cabestan, font=("Segoe Ui", 12), bg="green", fg="white")
+button_start.grid(row=0, column=0, padx=5)
+
+# Creation of red Stop button
+button_stop = tk.Button(cabestan_frame, text="Stop", command=stop_cabestan, font=("Segoe Ui", 12), bg="red", fg="white")
+button_stop.grid(row=0, column=1, padx=5)
+
+# Printing output speed of cabestan
+
+speed_cabestan = tk.Label(cabestan_frame, text="Speed:")
+speed_cabestan.grid(row=1, column=0, padx=5, columnspan=2)
+
+## Parameter frame section 
+parameter_frame = tk.LabelFrame(root, text="Parameters", height=100,width=150)
+parameter_frame.grid(row=2, column=4, rowspan=3, columnspan=3, padx=5, pady=5)
+
+# printing diameter measured by laser sensor
+diameter = tk.Label(parameter_frame, text="Diameter Measurement:")
+diameter.grid(row=0, column=0,columnspan=4, padx=5)
+
+# input for the diameter desired
+diameter_desired = tk.Label(parameter_frame, text='Diameter desired :')
+diameter_desired.grid(row=1, column=0)
+diameter_entry = tk.Entry(parameter_frame)
+diameter_entry.grid(row=1, column=1)
+diameter_entry_button = tk.Button(parameter_frame, text = 'Send', command=send_diameter)
+diameter_entry_button.grid(row=1, column = 2)
+
+
+## Debug section
+#Debug screen with time delay and received serial packets
+debug_frame = tk.LabelFrame(root, text="Debug", height=100,width=150)
+debug_frame.grid(row=5, column=4, rowspan=3, columnspan=3, padx=5, pady=5)
+serial_print = tk.Label(debug_frame, text="Serial")
+serial_print.grid(row=1, column=3, padx=5)
+time_print = tk.Label(debug_frame, text="Time")
+time_print.grid(row=1, column=3, padx=5)
+
+
+##Connection frame section
+connection_frame = tk.LabelFrame(root, text="Connection (WIP)", height=50,width=150)
+connection_frame.grid(row=0, column=4, rowspan=2, columnspan=2, padx=5, pady=5)
+
+# Port selection menu
+
+connect_button = tk.Button(connection_frame, text = "Reconnect", command=reconnect)
+connect_button.grid(row=0, column=1 , padx=5)
+
+connect_button = tk.Button(connection_frame, text = "Check Ports", command=check_ports)
+connect_button.grid(row=1, column=1 , padx=5)
+
+status_label = tk.Label(connection_frame, text = "Status: Disconnected", bg = 'red')
+status_label.grid(row=1, column=0 , padx=5)
+
+current_port = tk.StringVar()
+
+#If COM4 is available, it picks it as default when lauching or else it picks the first one available
+if "COM4" in serial_ports():
+    current_port.set("COM4") 
+else:
+    current_port.set(serial_ports()[0]) 
+
+connection_drop_menu = tk.OptionMenu(connection_frame, current_port, *serial_ports())
+connection_drop_menu.grid(row=0, column=0,columnspan=1, padx=5)
+
+
+def checkSerialPort():
+    """Check serial input from the arduino and put the value back into the corresponding label. 
+    String arriving from the arduino has the format "value1, value2, etc"
+    """
+    global timer
```

Comment:
global again. Time.time() could be an option.

Reply:
Removed, it wasn't that useful anymore